File: typescript-client-sdk/src/sdk/models/shared/codatpublicapimodelsclientsclientsyncsettingssingleputmodel.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { Expose, Transform } from "class-transformer";


export class CodatPublicApiModelsClientsClientSyncSettingsSinglePutModel extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "fetchOnFirstLink" })
  fetchOnFirstLink?: boolean;

  @SpeakeasyMetadata()
  @Expose({ name: "monthsToSync" })
  monthsToSync?: number;

  @SpeakeasyMetadata()
  @Expose({ name: "syncFromUtc" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  syncFromUtc?: Date;

  @SpeakeasyMetadata()
  @Expose({ name: "syncFromWindow" })
  syncFromWindow?: number;

  @SpeakeasyMetadata()
  @Expose({ name: "syncSchedule" })
  syncSchedule?: number;
}

File: go-client-sdk/pkg/models/shared/codatpublicapimodelsclientsclientsyncsettingssingleputmodel.go
package shared

import (
	"time"
)

type CodatPublicAPIModelsClientsClientSyncSettingsSinglePutModel struct {
	FetchOnFirstLink *bool      `json:"fetchOnFirstLink,omitempty"`
	MonthsToSync     *int       `json:"monthsToSync,omitempty"`
	SyncFromUtc      *time.Time `json:"syncFromUtc,omitempty"`
	SyncFromWindow   *int       `json:"syncFromWindow,omitempty"`
	SyncSchedule     *int       `json:"syncSchedule,omitempty"`
}
